src/useMultiRootEditor.tsx

```diff
@@ -0,0 +1,508 @@
+			// It may happen during the tests that the watchdog instance is not assigned before destroying itself. See: #197.
+			//
+			// Additionally, we need to find a way to detect if the whole context has been destroyed. As `componentWillUnmount()`
+			// could be fired by <CKEditorContext /> and <CKEditor /> at the same time, this `setTimeout()` makes sure
+			// that <CKEditorContext /> component will be destroyed first, so during the code execution
+			// the `ContextWatchdog#state` would have a correct value. See `EditorWatchdogAdapter#destroy()` for more information.
+			/* istanbul ignore next */
+			setTimeout( async () => {
+				if ( watchdog ) {
+					await watchdog.destroy();
+					watchdog = null;
+
+					return resolve();
+				}
+
+				if ( editor ) {
+					await editor.destroy();
+
+					return resolve();
+				}
+
+				resolve();
+			} );
+		} );
+	};
+
+	/**
+	 * Initializes the editor by creating a proper watchdog and initializing it with the editor's configuration.
+	 */
+	const _initializeEditor = async (): Promise<void> => {
+		if ( props.disableWatchdog ) {
+			await _createEditor( props.content, _getConfig() );
+
+			return;
+		}
+
+		/* istanbul ignore next */
+		if ( watchdog ) {
+			return;
+		}
+
+		if ( context instanceof ContextWatchdog ) {
+			watchdog = new EditorWatchdogAdapter( context );
+		} else {
+			watchdog = new EditorWatchdog( props.editor, props.watchdogConfig );
+		}
+
+		watchdog.setCreator( ( data, config ) => _createEditor( data as Record<string, string>, config ) );
+
+		watchdog.on( 'error', ( _, { error, causesRestart } ) => {
+			const onError = props.onError || console.error;
+
+			onError( error, { phase: 'runtime', willEditorRestart: causesRestart } );
+		} );
+
+		await watchdog
+			.create( content as any, _getConfig() )
+			.catch( error => {
+				const onError = props.onError || console.error;
+
+				onError( error, { phase: 'initialization', willEditorRestart: false } );
+			} );
+	};
+
+	useEffect( () => {
+		if ( !editor ) {
+			return;
+		}
+
+		if ( !shouldUpdateEditor.current ) {
+			shouldUpdateEditor.current = true;
+
+			return;
+		}
+
+		const editorData = editor.getFullData();
+		const editorAttributes = editor.getRootsAttributes();
+
+		const {
+			addedKeys: newRoots,
+			removedKeys: removedRoots,
+			modifiedKeys: modifiedRoots
+		} = _getStateDiff( editorData, content || {} );
+
+		const {
+			addedKeys: newAttributes,
+			removedKeys: removedAttributes,
+			modifiedKeys: modifiedAttributes
+		} = _getStateDiff( editorAttributes, attributes || {} );
+
+		const newRootAttributes = newAttributes.filter( rootAttr =>
+			!newRoots.includes( rootAttr ) && attributes[ rootAttr ] !== undefined );
+		const removedRootAttributes = removedAttributes.filter( rootAttr =>
+			!removedRoots.includes( rootAttr ) && attributes[ rootAttr ] !== undefined );
+		const modifiedRootAttributes = modifiedAttributes.filter( rootAttr => attributes[ rootAttr ] !== undefined );
```

Comment:
Do I get it correctly, that a proper name for `rootAttr` would be `rootName`?

Reply:
Yes. Fixed.